add check git repository method

georgefs opened the pull request:
#1

今天本來要去朝聖的>"<  

下大雨..(超沒誠意XD)

georgefs committed:
add check git repository method

moskytw commented on gitcount.py:
```diff
@@ -7,6 +7,11 @@
 from datetime import date, timedelta
 
+def check_git_resp():
+    p = Popen("git branch", shell=True, stdout=PIPE, stderr=PIPE)
```
You can use the `shell` function. :)

And why you use `git branch`? Isn't there any better approach?

georgefs replied:
這部份是經驗法則..
我有考慮過git log, branch, tag, status..

以使用上來說git branch的錯誤機率最小.. msg 也最少, 所以才使用git branch的

moskytw replied:
I don't buy it.

You should Google it, and you will find lot of better solutions. `git branch` does other things rather than just checking it is a git repos or not.

moskytw commented on gitcount.py:
```diff
@@ -53,6 +58,7 @@ def count(author=None, period='weekly', first='monday', number=None, range='', p
     The other arguments will be passed to the command, ``git log``.
     '''
 
+    assert check_git_resp(), 'git repository error'
```
This msg is not meaningful enough.

georgefs replied:
恩..  我只是覺得這樣比較乾淨, 所以把原本git 的error msg 都吃光光的"

moskytw replied:
I meant "git repository error" is not exact for this case.

If your function checks it is a git repository or not, why not just show "not a git repository"? It is clearer to user.

georgefs committed:
update by pull mosky request

update check git repository method, use git rev-parse which command is use on check git flags

change check_git_resp error msg to default git error msg

moskytw commented:
I fixed the issue you want to resolve. You can see this commit: https://github.com/moskytw/git-count/commit/b96325940e5e781680edaa79ad0d64bfb0dc2150.
